tests/proquest/test_importer.py

```diff
@@ -839,64 +842,39 @@ def test_fulfil_refreshes_expired_token(self):
             eq_(2, api_client_mock.get_book.call_count)
 
+    def test_correctly_imports_covers(self):
+        # We want to make sure that ProQuestOPDS2Importer
+        # correctly processes cover links in the ProQuest feed
+        # and generates LinkData for both, the full cover and thumbnail.
 
-PROQUEST_PUBLICATION_1 = OPDS2Publication(
-    metadata=PresentationMetadata(
-        identifier="urn:proquest.com/document-id/1",
-        modified=datetime.datetime(2020, 1, 31, 0, 0, 0),
-    )
-)
-
-PROQUEST_PUBLICATION_2 = OPDS2Publication(
-    metadata=PresentationMetadata(
-        identifier="urn:proquest.com/document-id/2",
-        modified=datetime.datetime(2020, 1, 30, 0, 0, 0),
-    )
-)
+        # Act
+        importer = ProQuestOPDS2Importer(self._db, self._proquest_collection)
 
-PROQUEST_PUBLICATION_3 = OPDS2Publication(
-    metadata=PresentationMetadata(
-        identifier="urn:proquest.com/document-id/3",
-        modified=datetime.datetime(2020, 1, 29, 0, 0, 0),
-    )
-)
+        result = importer.extract_feed_data(fixtures.PROQUEST_RAW_FEED)
 
-PROQUEST_PUBLICATION_4 = OPDS2Publication(
-    metadata=PresentationMetadata(
-        identifier="urn:proquest.com/document-id/4",
-        modified=datetime.datetime(2020, 1, 28, 0, 0, 0),
-    )
-)
+        # Assert
+        eq_(2, len(result))
+        publication_metadata_dictionary = result[0]
 
-PROQUEST_FEED_PAGE_1 = OPDS2Feed(
-    metadata=OPDS2FeedMetadata(
-        title="Page # 1", current_page=1, items_per_page=10, number_of_items=20
-    ),
-    groups=CollectionList(
-        [
-            OPDS2Group(
-                publications=CollectionList(
-                    [PROQUEST_PUBLICATION_1, PROQUEST_PUBLICATION_2]
-                )
-            )
+        eq_(
+            True,
+            fixtures.PROQUEST_RAW_PUBLICATION_ID in publication_metadata_dictionary,
+        )
+        publication_metadata = publication_metadata_dictionary[
+            fixtures.PROQUEST_RAW_PUBLICATION_ID
         ]
-    ),
-)
 
-PROQUEST_FEED_PAGE_2 = OPDS2Feed(
-    metadata=OPDS2FeedMetadata(
-        title="Page # 2", current_page=2, items_per_page=10, number_of_items=20
-    ),
-    groups=CollectionList(
-        [
-            OPDS2Group(
-                publications=CollectionList(
-                    [PROQUEST_PUBLICATION_3, PROQUEST_PUBLICATION_4]
-                )
-            )
-        ]
-    ),
-)
+        eq_(2, len(publication_metadata.links))
+
+        full_cover_link = publication_metadata.links[0]
+        eq_(True, isinstance(full_cover_link, LinkData))
+        eq_(fixtures.PROQUEST_RAW_PUBLICATION_COVER_HREF, full_cover_link.href)
+        eq_(Hyperlink.IMAGE, full_cover_link.rel)
+
+        thumbnail_cover_link = publication_metadata.links[1]
+        eq_(True, isinstance(thumbnail_cover_link, LinkData))
+        eq_(fixtures.PROQUEST_RAW_PUBLICATION_COVER_HREF, thumbnail_cover_link.href)
+        eq_(Hyperlink.THUMBNAIL_IMAGE, thumbnail_cover_link.rel)
```

Comment:
Yes, this only tests that the LinkData objects were created. I think if you call `metadata.apply` in this situation you won't get the results you want, because the image and the thumbnail aren't connected.

Reply:
I see, the end result is still valid, even though the work's record in the database is in an inconsistent state. I'm glad that works but yes, it would be better to have the image associated with its thumbnail.